src/core/include/mp-units/cartesian_vector.h

```diff
@@ -203,13 +232,11 @@ class cartesian_vector : public detail::cartesian_vector_iface {
   [[nodiscard]] constexpr cartesian_vector operator+() const { return *this; }
   [[nodiscard]] constexpr cartesian_vector operator-() const
   {
-    return {-_coordinates_[0], -_coordinates_[1], -_coordinates_[2]};
+    return cartesian_vector{-_coordinates_[0], -_coordinates_[1], -_coordinates_[2]};
   }
 
   template<typename U>
-    requires requires(T t, U u) {
-      { t += u } -> std::same_as<T&>;
-    }
+    requires requires(T& t, const U& u) { t += u; }
```

Comment:
Why did you drop the return type check? If it is not satisfied, then the code will not compile.